src/containerapp/azext_containerapp/custom.py

```diff
@@ -1881,3 +1884,100 @@ def init_dapr_components(cmd, resource_group_name, environment_name, statestore=
             "daprComponents": [statestore_component_id, pubsub_component_id]
         }
     }
+
+
+def list_java_components(cmd, environment_name, resource_group_name):
+    raw_parameters = locals()
+    java_component_decorator = JavaComponentDecorator(
+        cmd=cmd,
+        client=JavaComponentPreviewClient,
+        raw_parameters=raw_parameters,
+        models=CONTAINER_APPS_SDK_MODELS
+    )
+    return java_component_decorator.list()
+
+
+def show_java_component(cmd, java_component_name, environment_name, resource_group_name, target_java_component_type):
+    raw_parameters = locals()
+    java_component_decorator = JavaComponentDecorator(
+        cmd=cmd,
+        client=JavaComponentPreviewClient,
+        raw_parameters=raw_parameters,
+        models=CONTAINER_APPS_SDK_MODELS
+    )
+    result = java_component_decorator.show()
+
+    current_type = safe_get(result, "properties", "componentType")
+    if current_type and target_java_component_type.lower() != current_type.lower():
+        raise ResourceNotFoundError(f"(JavaComponentNotFound) JavaComponent '{java_component_name}' was not found.")
+
+    return result
+
+
+def delete_java_component(cmd, java_component_name, environment_name, resource_group_name, target_java_component_type, no_wait):
+    show_java_component(cmd, java_component_name, environment_name, resource_group_name, target_java_component_type)
```

Comment:
Is it required to show the java component before execute deletion?

The results of multiple executions of the `delete` command be consistent(idempotent).
If delete a not existing java component, the result should be success, right?

Reply:
I have updated the logic. The background here is that if there is a java component, and its type is `config server`, and its name is `configserver01`. In this case, if customer use the cli command az containerapp env java-component spring-cloud-**eureka** to delete the java component, we will tell customers that the target resource not found. If they use az containerapp env java-component spring-cloud-config, they can delete it successfully.

So, the new logic here is that I will check whether the target resource exist. If not, CLI will return success, if exist, CLI will check the component type.